test/CMakeLists.txt

```diff
@@ -1,12 +1,23 @@
+IF (NOT WIN32)
+    SET (RUNTESTCASES_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/runtestcases.sh)
+ELSE ()
+    SET (RUNTESTCASES_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/runtestcases.bat)
+ENDIF ()
+
 FOREACH(tcdir testcases libsolv-zypptestcases)
     IF(IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/${tcdir}")
         FILE(GLOB dirs "${CMAKE_CURRENT_SOURCE_DIR}/${tcdir}/[_a-zA-Z0-9]*")
         FOREACH(dir ${dirs})
-	    IF(IS_DIRECTORY ${dir})
-		FILE(RELATIVE_PATH myname "${CMAKE_CURRENT_SOURCE_DIR}/${tcdir}" ${dir})
-		STRING(REPLACE \\ \\\\ dir ${dir})
-		ADD_TEST(${myname} ${CMAKE_CURRENT_SOURCE_DIR}/runtestcases ${CMAKE_BINARY_DIR}/tools/testsolv ${dir})
-	    ENDIF(IS_DIRECTORY ${dir})
-        ENDFOREACH(dir)
-    ENDIF(IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/${tcdir}")
-ENDFOREACH(tcdir)
+            IF(IS_DIRECTORY ${dir})
+                FILE(RELATIVE_PATH myname "${CMAKE_CURRENT_SOURCE_DIR}/${tcdir}" ${dir})
+                FILE(TO_CMAKE_PATH "${CMAKE_BINARY_DIR}/tools/testsolv" TESTSOLV_BINARY)
+                FILE(TO_CMAKE_PATH "${dir}" dir)
+                if (WIN32)
+                    STRING(REGEX REPLACE "/" "\\\\" dir ${dir})
+                    STRING(REGEX REPLACE "/" "\\\\" TESTSOLV_BINARY ${TESTSOLV_BINARY})
+                endif ()
```

Comment:
1) You deleted my `STRING(REPLACE \\ \\\\ dir ${dir})` line. It is also needed for non-win32 systems.
2) your regex replaces are overkill. You should simply use `FILE(TO_NATIVE_PATH` instead. The \ -> \\ replace will then give you the desired result
3) do you really need the `FILE(TO_CMAKE_PATH` calls? Isn't the `FILE(GLOB` result already a cmake path?

Reply:
I simplified it, warnings are gone :+1: